server.go

```diff
@@ -126,6 +132,24 @@ func (u *Upgrader) Upgrade(w http.ResponseWriter, r *http.Request, responseHeade
 	}
 
 	subprotocol := u.selectSubprotocol(r, responseHeader)
+	if responseHeader == nil {
+		responseHeader = http.Header{}
+	} else {
+		delete(responseHeader, "Sec-Websocket-Extensions")
```

Comment:
Do not modify the application's headers.  Follow the pattern used for Sec-Websocket-Protocol.